### ISSUE TITLE
Everytime I start Steam it "fossilize replay"

### ISSUE THREAD

**Issue by Zorrototo** [ValveSoftware/steam-for-linux#7443]
#### Your system information

* Steam client version (build number or date): Oct 23 2020, at  20:26:58
* Distribution (e.g. Ubuntu): Manjaro KDE
* Opted into Steam client beta?: [No]
* Have you checked for system updates?: [Yes]

#### Please describe your issue in as much detail as possible:
When I start Steam, for which I enabled shader precaching and allowe backgroung processing of shaders, every single time, it goes through the shaders and process them in the background. Nothing has changed about those shaders, I didn't play the games, I did not download any shader update or anything. Despite it shouldn't always process again and again the shaders, if I restart the computer and start Steam, then it goes again for validating and fossilizing replay in the background.
Also yesterday I added swap to my computer, I never crash because of lack of swap or RAM, but the fossilize process doesn't care and fills the 16GB RAM and had crashed the computer before I add SWAP (systemd-swap), so here I am, needed SWAP for Steam to not crash my system..

#### Steps for reproducing this issue:

1. Restart the computer
2. Start Steam
3. Notice your full speed CPU fan speed for 15 minutes and notice in HTOP tool the CPU and RAM usage comes from Steam's fossilize_replay tool, which seem to do the same job every time for no reason, as it previously did that on previous boot or previous start of Steam, when there was no shader cache download, and no play session for the affected games shaders processed every time I reboot (for instance Rocket League).

**Comment by Zorrototo** [comment]
I restarted after writing this, and of course it doesn't "fossilize_replay" this time, discarding my whole post. BUT I will continue observing and logging this behavior from now on.

- November 1st, 11:15 it "fossilize_replay" for Rocket League shaders filling the whole 16GB RAM at some point freezing the system, thank to systemd-swap for adjusting itself to avoid crash. No prior shader cache update for this game.

**Comment by kisak-valve** [comment]
Hello @Zorrototo, this is the same issue as #7306 and https://github.com/ValveSoftware/Fossilize/issues/84. Closing as a duplicate of the existing issue report.

**Comment by jarrard** [comment]
Updated to latest Plasma desktop and it seems steam is redoing all shaders. 
I presume it will eventually be done, but atm it does 3 processes of 8% CPU utilization with fossilized_replay. 
They generally only run for a few seconds but can stick around for a few minutes before opening a new set of 3 processes.

This is likely steam going through my library 1 game at a time and generating new shaders, and it should end eventually. (I hope)

**Comment by Zorrototo** [comment]
Issue is fixed since longtime on Steam BETA. Also as pointed above this issue here is closed, discussion goes on the other issue linked.

**Comment by jarrard** [comment]
Yes, I just wanted to point out what people might see with the process sticking around for a while. BUT it does eventually finish doing its fossilized stuff and all goes back to normal.

**Comment by paulmercier-bisq** [comment]
Disabling the "Allow vulkan shader background processing" in Settings>Shader Pre-caching seems to fix it.

**Comment by Zorrototo** [comment]
Issue was fixed multiple months ago. Also people continue replying to this closed issue instead of the open one..

**Comment by hartescout** [comment]
> Disabling the "Allow vulkan shader background processing" in Settings>Shader Pre-caching seems to fix it.

Seems to be the only fix for me. I'm running PopOS 22.04 updated. Flatpak install.

**Comment by mikkorantalainen** [comment]
Disabling the "Allow vulkan shader background processing" in Settings - Download seemed to help the issue for me. I still had to let it process the shaders once with a single run (took about 15 minutes) but then it was fine until the next the software was updated.

I'm still wondering why Steam doesn't have incremental mode where it could continue from the already processed part if the process is interrupted. And if you accidentally click the window, it's the same as "Cancel" which obviously throws away all the computation done that far.

**Comment by sabo-tabby-kitten** [comment]
this is because steam is downloding volken shaders in the backround you stop it by turning off the setting in comatibility

**Comment by rebroad** [comment]
This feature seems to be problematic - two processes both using up 100% CPU running concurrently with any games - making the games unplayable due to lack of CPU.

**Comment by hyperlogos** [comment]
> This feature seems to be problematic - two processes both using up 100% CPU running concurrently with any games - making the games unplayable due to lack of CPU.

I wish I only had 2 processes doing this. I have six.

**Comment by Zorrototo** [comment]
Shader processing should not occur during gameplay. It should happen in the background or BEFORE starting a game. If it processes during gameplay you probably are doing something not conventional.

**Comment by derbolle** [comment]
@hyperlogos you can create the file `~/.steam/steam/steam_dev.cfg` and put the following there: 

`unShaderBackgroundProcessingThreads 2` 

Then it should use only 2 threads. Hope this helps

**Comment by hyperlogos** [comment]
> Then it should use only 2 threads. Hope this helps

I'll give it a try, thanks for the tip. I wish I could just drop the priority and iopriority. _nice ionice -c 3 command_ will allow a process to run without beating up other processes. Though since there's multiple processes, it would probably make more sense to use cgroups.
